src/framework/CMakeLists.txt

```diff
@@ -217,57 +217,67 @@ message(STATUS "LuaJIT: " ${LUAJIT})
 
 find_package(PhysFS REQUIRED)
 find_package(ZLIB REQUIRED)
-find_package(Protobuf REQUIRED)
-find_package(LibLZMA REQUIRED)
-find_package(nlohmann_json REQUIRED)
 
 set(framework_LIBRARIES ${framework_LIBRARIES}
     ${Boost_LIBRARIES}
     ${LUA_LIBRARY}
     ${PHYSFS_LIBRARY}
     ${ZLIB_LIBRARY}
-    ${PROTOBUF_LIBRARY}
-    ${LIBLZMA_LIBRARIES}
-    ${NLOHMANN_JSON_LIBRARY}
 )
 
 set(framework_INCLUDE_DIRS ${framework_INCLUDE_DIRS}
     ${Boost_INCLUDE_DIRS}
     ${LUA_INCLUDE_DIR}
     ${PHYSFS_INCLUDE_DIR}
-    ${framework_INCLUDE_DIRS}
-    ${PROTOBUF_INCLUDE_DIRS}
-    ${LIBLZMA_INCLUDE_DIRS}
-    ${NLOHMANN_JSON_INCLUDE_DIR}
 )
 
-# protobuf
-set(protobuf_PATH ${CMAKE_CURRENT_LIST_DIR}/protobuf/proto)
-set(protobuf_MODULES
-    appearances
-    shared
-)
+if(USE_PROTOBUF)
+    find_package(Protobuf REQUIRED)
+    find_package(LibLZMA REQUIRED)
+    find_package(nlohmann_json REQUIRED)
 
-foreach(protobuf_FILENAME ${protobuf_MODULES})
+    set(framework_LIBRARIES ${framework_LIBRARIES}
+        ${PROTOBUF_LIBRARY}
+        ${LIBLZMA_LIBRARIES}
+        ${NLOHMANN_JSON_LIBRARY}
+    )
 
-    set(protobuf_SOURCES
-        ${CMAKE_CURRENT_LIST_DIR}/protobuf/${protobuf_FILENAME}.pb.cc
-        ${CMAKE_CURRENT_LIST_DIR}/protobuf/${protobuf_FILENAME}.pb.h
+    set(framework_INCLUDE_DIRS ${framework_INCLUDE_DIRS}
+        ${PROTOBUF_INCLUDE_DIRS}
+        ${LIBLZMA_INCLUDE_DIRS}
+        ${NLOHMANN_JSON_INCLUDE_DIR}
     )
 
-    set(framework_SOURCES ${framework_SOURCES} ${protobuf_SOURCES})
+    # protobuf
+    set(protobuf_PATH ${CMAKE_CURRENT_LIST_DIR}/protobuf/proto)
+    set(protobuf_MODULES
+        appearances
+        shared
+    )
 
-    if(NOT WIN32)
-        add_custom_command(
-            OUTPUT ${protobuf_SOURCES}
-            DEPENDS ${protobuf_PATH}/${protobuf_FILENAME}.proto
-            COMMAND protoc --cpp_out=../ ${protobuf_FILENAME}.proto
-            WORKING_DIRECTORY ${protobuf_PATH}
-            COMMENT "Compiling ${protobuf_FILENAME}.proto"
+    foreach(protobuf_FILENAME ${protobuf_MODULES})
+
+        set(protobuf_SOURCES
+            ${CMAKE_CURRENT_LIST_DIR}/protobuf/${protobuf_FILENAME}.pb.cc
+            ${CMAKE_CURRENT_LIST_DIR}/protobuf/${protobuf_FILENAME}.pb.h
         )
-    endif()
 
-endforeach()
+        set(framework_SOURCES ${framework_SOURCES} ${protobuf_SOURCES})
+
+        if(NOT WIN32)
+            add_custom_command(
+                OUTPUT ${protobuf_SOURCES}
+                DEPENDS ${protobuf_PATH}/${protobuf_FILENAME}.proto
+                COMMAND protoc --cpp_out=../ ${protobuf_FILENAME}.proto
+                WORKING_DIRECTORY ${protobuf_PATH}
+                COMMENT "Compiling ${protobuf_FILENAME}.proto"
+            )
+        endif()
+
+    endforeach()
+    set(framework_DEFINITIONS ${framework_DEFINITIONS} -DUSE_PROTOBUF)
```

Comment:
You moved all protobuf files to client source folder (not sure why), but this snippet is trying to compile them from the framework folder...
```
make[2]: *** No rule to make target '../src/framework/protobuf/proto/shared.proto', needed by '../src/framework/protobuf/shared.pb.cc'.  Stop.
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/otclient.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

```

Reply:
Aight will fix this